compiler/block.py

```diff
@@ -440,16 +440,29 @@ def __init__(self, node):
     BlockVisitor.__init__(self)
     self.is_generator = False
     node_args = node.args
-    args = [a.arg for a in node_args.args]
+    args = []
+    for arg in node_args.args:
+      if isinstance(arg, ast.Tuple):
+        args.append([a.arg for a in arg.elts])
+      else:
+        args.append(arg.arg)
     if node_args.vararg:
       args.append(node_args.vararg.arg)
     if node_args.kwarg:
       args.append(node_args.kwarg.arg)
     for i, name in enumerate(args):
-      if name in self.vars:
-        msg = "duplicate argument '{}' in function definition".format(name)
-        raise util.ParseError(node, msg)
-      self.vars[name] = Var(name, Var.TYPE_PARAM, arg_index=i)
+      if isinstance(name, list):
+        self.add_vars(i, 'τ'.join(name), node)
```

Comment:
As it turns out the arg tuples can be nested:

```
>>> def f((a, (b, c))):
...  print a, b, c
... 
>>> f((1, (2, 3)))
1 2 3
```

This points to a recursive approach similar to the way [_tie_target()](https://github.com/google/grumpy/blob/master/compiler/stmt.py#L751) in stmt.py works. My intuition is that we can use the exact same mechanism (grumpy.TieTarget struct) for unpacking the args.

Also, Var represents Python variables and are used to resolve names and handle scoping rules so I don't think you want to create a pseudo Var to represent the tuple. Instead you want to put the special unpacking logic somewhere around [here](https://github.com/google/grumpy/blob/master/compiler/expr_visitor.py#L464). But instead of assigning to each of the locals for the args, that's where you want to do the unpacking use the TieTarget mechanism.